contracts/core/zapper/WidoZapperVelodrome.sol

```diff
@@ -134,4 +123,123 @@ contract WidoZapperVelodrome is WidoZapperUniswapV2 {
         );
         amountOut = amounts[1];
     }
+
+    /// @inheritdoc WidoZapperUniswapV2
+    function _feeBps(
+        IUniswapV2Router02 router,
+        IUniswapV2Pair pair,
+        bool //isFromToken0
+    ) internal pure virtual override returns (uint256) {
+        VelodromePairFactory factory = VelodromePairFactory(VelodromeRouter(address(router)).factory());
+        return factory.getFee(VelodromePair(address(pair)).stable());
+    }
+
+    /// @inheritdoc WidoZapperUniswapV2
+    function _getAmountOut(
+        IUniswapV2Router02 router,
+        IUniswapV2Pair pair,
+        uint256 amountIn,
+        Asset memory assetIn,
+        Asset memory assetOut,
+        bytes memory //extra
+    )
+    internal view virtual override
+    returns (uint256 amountOut) {
+        VelodromePairFactory factory = VelodromePairFactory(VelodromeRouter(address(router)).factory());
+        bool stable = VelodromePair(address(pair)).stable();
+        uint reserve0;
+        uint reserve1;
+
+        if (VelodromePair(address(pair)).token0() == assetIn.token){
+            reserve0 = assetIn.reserves;
+            reserve1 = assetOut.reserves;
+        }
+        else {
+            reserve0 = assetOut.reserves;
+            reserve1 = assetIn.reserves;
+        }
+
+        amountIn -= amountIn * factory.getFee(stable) / 10000; // remove fee from amount received
```

Comment:
Should we use FEE_DENOMINATOR here?

Reply:
no, this is the fee in the "Velodrome format", so we have to keep their denominator. It happens to be the same of ours... but i think it could confuse to use ours. Will add comment.